packages/client-config/src/model_introspection_schema_adapter.ts

```diff
@@ -18,9 +20,28 @@ export class ModelIntrospectionSchemaAdapter {
   getModelIntrospectionSchemaFromS3Uri = async (
     modelSchemaS3Uri: string | undefined
   ): Promise<unknown | undefined> => {
-    return getModelIntrospectionSchemaFromS3Uri({
-      credentialProvider: this.awsCredentialProvider,
-      modelSchemaS3Uri,
-    });
+    if (!modelSchemaS3Uri) {
+      return;
+    }
+    const generatedModels = await (
+      await createGraphqlModelsGenerator({
+        modelSchemaS3Uri,
+        credentialProvider: this.awsCredentialProvider,
+      }).generateModels({ target: 'introspection' })
+    ).getResults();
+    const generatedModelFiles = Object.values(generatedModels);
+    if (generatedModelFiles.length !== 1) {
+      throw new Error(
+        `A single model introspection schema is expected, received ${generatedModelFiles.length} values.`
+      );
```

Comment:
Change to AmplifyError

Reply:
sounds good. 
I converted these to faults. These should not happen under normal circumstances nor could be result of normal user actions.